cfitsio3470/docs/changes.txt

```diff
@@ -1,4 +1,93 @@
+  - fits_copy_rows (ffcprw) can now handle 'P'-type variable-length 
+    columns.
+    
+  - Fix made to an obscure case in fits_modify_vector_len, where a 
+    wrongly issued EOF error may occur.
+                   
+  - Added internal fffvcl() function.
+                   
+Version 3.46 - Oct 2018 (Ftools release)
+                   
+  - Improved the algorithm for ensuring no tile dimensions are smaller 
+    than 4 pixels for HCOMPRESS compression.
+                   
+  - Added new functions intended to assist in diagnosing (primarily 
+    https) download issues: fits_show_download_progress,
+    fits_get_timeout, fits_set_timeout.
+
+  - Added the '-O <file>' option to fpack, which previously existed only
+    for funpack.  Also added fpack/funpack auto-removal of .bz2 suffix 
+    equivalent to what existed for .gz.
+
+  - For the fpack '-table' cases, warning message is now sent to stderr 
+    instead of stdout.  This is to allow users to pipe the results from
+    stdout in valid FITS format.  (The warning message is otherwise placed
+    at the start of the FITS file and therefore corrupts it.)
+
+  - Fix made to the '-P' file prefix option in funpack.
+
+  - Added wildcard deletion syntax for columns, i.e. -COLNAM* will delete
+    the first matching column as always; -COLNAM*+ will delete all matching
+    columns (or none); exact symmetry with the keyword deletion syntax.
+
+Version 3.45 - May 2018
+                   
+  - New support for reading and writing unsigned long long datatypes.
+    This includes 'implicit datatype conversion' between the unsigned long
+    long datatype and all the other datatypes.
+    
+  - Increased the hardcoded NMAXFILES setting for maximum number of
+    open files from 1000 to 10000.
+                   
+  - Bug fix to fits_calc_binning wrapper function, which wasn't filling
+    in the returned float variables.
+    
+  - Fixed a parsing bug for image subsection and column binning range
+    specifiers that was introduced in v3.44.
+
+Version 3.44 - April 2018
+
+  - This release primarily patches security vulnerabilities.  We
+    strongly encourage this upgrade, particularly for those running 
+    CFITSIO in web accessible applications.
+    
+    In addition, the following enhancements and fixes were made: 
+
+  - Enhancement to 'template' and 'colfilter' functionality.  It is now
+    possible to delete multiple keywords using wildcard syntax. See
+    "Column and Keyword Filtering Specification" section of manual for
+    details.
+  
+  - histo.c uses double precision internally for all floating point
+    binning; new double-precision subroutines fits_calc_binningd(),
+    fits_rebin_wcsd(), and fits_make_histd(); existing
+    single-precision histogram functions still work but convert values
+    to double-precision internally.
+
+  - new subroutine fits_copy_cols() / ffccls() to copy multiple columns
+  
+  - Fix in imcompress.c for HCOMPRESS and PLIO compression of unsigned
```

Comment:
There is a patch for this bug in particular that may need to be removed.

Reply:
OK.  Does the Travis build fail if the patches fail?

Reply:
Unclear. You will need to go through the patches by hand and check the source to make sure they still work. These patches are the least-worst solution.